specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2022-06-01-preview/loadtestservice.json

```diff
@@ -0,0 +1,2490 @@
+            "type": "string",
+            "format": "date-time"
+          },
+          {
+            "in": "query",
+            "name": "continuationToken",
+            "description": "Continuation token to get the next page of response.",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "maxPageSize",
+            "description": "Number of results in response.",
+            "type": "integer",
+            "format": "int32",
+            "default": 50
+          },
+          {
+            "$ref": "#/parameters/ApiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The requested load tests",
+            "schema": {
+              "$ref": "#/definitions/TestModelResourceList"
+            }
+          },
+          "default": {
+            "description": "Load Testing service error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseBody"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "The error code for specific error that occurred.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ListLoadTestSearch": {
+            "$ref": "./examples/ListLoadTestSearch.json"
+          }
+        }
+      }
+    },
+    "/loadtests/{testId}/files/{fileId}": {
+      "put": {
+        "tags": [
+          "Test"
+        ],
+        "summary": "Upload input file for a given test name. File size can't be more than 50 MB. Existing file with same name for the given test will be overwritten.",
+        "operationId": "Test_UploadTestFile",
+        "consumes": [
+          "multipart/form-data"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "fileId",
+            "description": "Unique identifier for test file, must be a valid URL character ^[a-z0-9_-]*$.",
+            "required": true,
+            "type": "string",
+            "maxLength": 50,
+            "minLength": 2,
+            "pattern": "^[a-z0-9_-]*$"
+          },
+          {
+            "in": "query",
+            "name": "fileType",
+            "type": "integer",
+            "format": "int32",
+            "description": "Integer representation of the file type (0 = JMX_FILE, 1 = USER_PROPERTIES, 2 = ADDITIONAL_ARTIFACTS)."
+          },
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/TestIdPathParameter"
+          },
+          {
+            "in": "formData",
+            "name": "file",
+            "required": true,
+            "type": "file",
+            "description": "The file to be uploaded."
+          }
```

Comment:
@mikekistler  If we use octet-stream then we will have to guide the user to pass the file name either in the URL or body or in the customer header. If we think of taking the file name in the URL param, then for the user it will be restricted with URL character and have to do encoding for other characters. Let us know if there is any better way to take handle this in octet-stream. In the case of multipart, it will be taken care of in content disposition automatically

Reply:
Good point.  I had not considered that.  It is reasonable to leave it as multipart/form-data then.